Detectors/TRD/base/include/TRDBase/TRDGeometry.h

```diff
@@ -11,203 +11,57 @@
 #ifndef O2_TRDGEOMETRY_H
 #define O2_TRDGEOMETRY_H
 
+#include "TRDBase/TRDGeometryBase.h"
+#include "DetectorsCommonDataFormats/DetMatrixCache.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+
 #include <string>
 #include <vector>
+#include <memory>
 
-class TGeoHMatrix;
+using namespace o2::trd;
 
 namespace o2
 {
 namespace trd
 {
-class TRDPadPlane;
 
-class TRDGeometry
+class TRDGeometry : public TRDGeometryBase, public o2::detectors::DetMatrixCacheIndirect
```

Comment:
any reason why public (double) inheritance is favoured over composition?
(an alternative may be protected or private inheritance)

Reply:
Because I want to expose all interfaces of TRDGeometryBase and of FlatObject to the outside.